gtda/utils/validation.py

```diff
@@ -77,102 +78,126 @@ def check_graph(X):
 
 # Check the type and range of numerical parameters
 def validate_params(parameters, references):
+    """Function to automate the validation of hyperparameters.
+
+    Parameters
+    ----------
+    parameters : dict, required
+        Dictionary in which the keys are hyperparameter names and the
+        corresponding values are hyperparameter values.
+
+    references : dict, required
+        Dictionary in which the keys are hyperparameter names and the
+        corresponding values are lists. The first element of that list is a
+        type and the second can be one of:
+
+        - ``None``, when only the type should be checked.
+        - A tuple of two numbers, when the type is numerical. In this case,
+          the first (resp. second) entry in the tuple defines a lower
+          (resp. upper) bound constraining the value of the corresponding
+          hyperparameter.
+        - A list containing all possible allowed values for the
```

Comment:
I don't like that two similar iterables (list and tuples) are interpreted completely differently. Do i understand correctly that 
`[list, (int, [0,1,2])]` and `[list, (int, (0,2))]` produce the same behavior?